core_impl/src/test/java/com/google/instrumentation/stats/StatsManagerImplTest.java

```diff
@@ -69,23 +82,109 @@ public void testGetNonexistentView() throws Exception {
     statsManager.registerView(RpcConstants.RPC_CLIENT_ROUNDTRIP_LATENCY_VIEW);
-    statsManager.record(
-        StatsContextFactoryImpl.DEFAULT,
-        MeasurementMap.of(RpcConstants.RPC_CLIENT_ROUNDTRIP_LATENCY, 10));
+    // Record stats 10.0, 20.0, 30.0, 40.0
+    for (int i = 1; i <= 4; i++) {
+      statsManager.record(new StatsContextImpl(oneTag),
+          MeasurementMap.of(RpcConstants.RPC_CLIENT_ROUNDTRIP_LATENCY, 10.0 * i));
+    }
+
     DistributionView view =
         (DistributionView) statsManager.getView(RpcConstants.RPC_CLIENT_ROUNDTRIP_LATENCY_VIEW);
     assertThat(view.getViewDescriptor()).isEqualTo(RpcConstants.RPC_CLIENT_ROUNDTRIP_LATENCY_VIEW);
-    assertThat(view.getDistributionAggregations()).hasSize(1);
-    assertThat(view.getDistributionAggregations().get(0).getCount()).isEqualTo(1);
-    assertThat(view.getDistributionAggregations().get(0).getSum()).isWithin(TOLERANCE).of(10.0);
-    assertThat(view.getDistributionAggregations().get(0).getMean()).isWithin(TOLERANCE).of(10.0);
+    assertThat(view.getEnd().getSeconds()).isAtLeast(view.getStart().getSeconds());
+    List<DistributionAggregation> distributionAggregations = view.getDistributionAggregations();
+    assertThat(distributionAggregations).hasSize(1);
+    DistributionAggregation distributionAggregation = distributionAggregations.get(0);
+    verifyDistributionAggregation(distributionAggregation,
+        RpcConstants.RPC_MILLIS_BUCKET_BOUNDARIES.size() + 1, 4,100.0,25.0,10.0,40.0,1);
+    List<Tag> tags = distributionAggregation.getTags();
+    assertThat(tags.get(0).getKey()).isEqualTo(tagKey);
+    assertThat(tags.get(0).getValue()).isEqualTo(tagValue1);
+  }
+
+  @Test
+  public void testRecordMultipleTagValues() {
+    statsManager.registerView(RpcConstants.RPC_CLIENT_ROUNDTRIP_LATENCY_VIEW);
+    statsManager.record(new StatsContextImpl(oneTag),
+        MeasurementMap.of(RpcConstants.RPC_CLIENT_ROUNDTRIP_LATENCY, 10.0));
+    statsManager.record(new StatsContextImpl(anotherTag),
+        MeasurementMap.of(RpcConstants.RPC_CLIENT_ROUNDTRIP_LATENCY, 30.0));
+    statsManager.record(new StatsContextImpl(anotherTag),
+        MeasurementMap.of(RpcConstants.RPC_CLIENT_ROUNDTRIP_LATENCY, 50.0));
+
+    DistributionView view =
+        (DistributionView) statsManager.getView(RpcConstants.RPC_CLIENT_ROUNDTRIP_LATENCY_VIEW);
+    List<DistributionAggregation> distributionAggregations = view.getDistributionAggregations();
+    assertThat(distributionAggregations).hasSize(2);
+    DistributionAggregation distributionAggregation1 = distributionAggregations.get(0);
+    DistributionAggregation distributionAggregation2 = distributionAggregations.get(1);
+
+    // The order of distributionAggregation1 and distributionAggregation2 could be random
+    if (distributionAggregation1.getCount() == 2) {
+      DistributionAggregation distributionAggregation = distributionAggregation1;
+      distributionAggregation1 = distributionAggregation2;
```

Comment:
Would it be simpler to copy distributionAggregations into an ArrayList and then sort it on count (with a Comparator) to give it a fixed order?  Then you wouldn't have to swap the variables.